packages/flutter_tools/lib/src/macos/cocoapod_utils.dart

```diff
@@ -25,13 +25,6 @@ Future<void> processPodsIfNeeded(
   if (project.usesSwiftPackageManager && !xcodeProject.podfile.existsSync() && !forceCocoaPodsOnly) {
     return;
   }
-  // Ensure that the plugin list is up to date, since hasPlugins relies on it.
-  await refreshPluginsList(
-    project,
-    iosPlatform: project.ios.existsSync(),
-    macOSPlatform: project.macos.existsSync(),
-    forceCocoaPodsOnly: forceCocoaPodsOnly,
```

Comment:
In https://github.com/flutter/flutter/issues/157391#issuecomment-2430471302 I said

> I don't think it was added in #146256, but the call site parameters were changed.

This was true, but I think the logic added in that PR was intended to `forceCocoaPodsOnly` for add-to-app, since add-to-app currently only supports CocoaPods and not Swift Package Manager.

https://github.com/flutter/flutter/blob/7504abcc14a6317e55a34ff7d9b9f9ce6be59750/packages/flutter_tools/lib/src/commands/build_ios_framework.dart#L263-L268

https://github.com/flutter/flutter/blob/7504abcc14a6317e55a34ff7d9b9f9ce6be59750/packages/flutter_tools/lib/src/flutter_plugins.dart#L207


However, [`build_ios_framework_module_test`](https://github.com/flutter/flutter/blob/7504abcc14a6317e55a34ff7d9b9f9ce6be59750/dev/devicelab/bin/tasks/build_ios_framework_module_test.dart) (post-submit only test) would test this, so I checked this out, set `flutter config --enable-swift-package-manager`, and it still passed.  Which makes sense because that `project.usesSwiftPackageManager` explicitly checks if it's an add-to-app module (`!isModule`).

https://github.com/flutter/flutter/blob/7504abcc14a6317e55a34ff7d9b9f9ce6be59750/packages/flutter_tools/lib/src/project.dart#L269-L272

All that to say, I still think this isn't needed, but not for the reason I said before.

@loic-sharma can you check my work here?

Reply:
> Which makes sense because that `project.usesSwiftPackageManager` explicitly checks if it's an add-to-app module (`!isModule`).

...and I just remembered that command can be run from a normal `flutter create`d iOS or macOS app, not just a module.

On this PR:
```
$ flutter config --enable-swift-package-manager
$ flutter create test_create
$ cd test_create
$ flutter pub add camera
$ flutter build ios-framework --xcframework --no-debug --no-profile
$ ls  build/ios/framework/Release/
```
On master that directory contains camera_avfoundation.xcframework, but on this PR it doesn't.  

`.flutter-plugins-dependencies` should have `swift_package_manager_enabled` false at this point.
```
"swift_package_manager_enabled":true
```

Reply:
@loic-sharma this would be a good integration test to add.

Reply:
The `flutter build ios-framework` command acts unexpectedly if Swift Package Manager is enabled. If I do this:

```
flutter config --enable-swift-package-manager
flutter build ios-framework --xcframework --no-debug --no-profile
```

The build completes, however, this unexpectedly generates a Swift package manifest that includes plugins and then later [deletes it](https://github.com/flutter/flutter/blob/0fd477d7f878cb7e8cbea0b61a06914a5a6c1a24/packages/flutter_tools/lib/src/macos/cocoapod_utils.dart#L54-L57).

Interestingly, the produced `App.xframework` doesn't seem to statically link against the plugins?

```
$ nm --extern-only --just-symbol-name build/ios/framework/Release/App.xcframework/ios-arm64/App.framework/App -arch arm64
_kDartIsolateSnapshotData
_kDartIsolateSnapshotInstructions
_kDartVmSnapshotData
_kDartVmSnapshotInstructions
dyld_stub_binder
```

I'll need to dig into this more

Reply:
Sounds good, thanks for looking at this!

I am not explicitly blocked, but I would like to know (can be tomorrow) if I should expect to pursue this change (that is, it _should_ be OK if other bugs or considerations are fixed) or if I should expect to drop it (in which case I might want to just opt-out all users of `processPodsIfNeeded` from generating `.flutter-plugins`: https://github.com/flutter/flutter/pull/157393#issuecomment-2433253573).

Reply:
I suspect you should drop this for now and reassign to me. I should clean this up as part of adding add-to-app support to SwiftPM. Thoughts @jmagman?